[Chua Ming Quan] iP

mqchua opened the pull request:
# Duke
> “Your personal todo reminder bot!” – Duke

Duke frees your mind of having to remember things you need to do. It's,

text-based
easy to learn
~~FAST~~ SUPER FAST to use

All you need to do is,

1 .download it from [here](https://github.com/mqchua/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is __FREE__!

Features:

- [x] Managing tasks
- [x] Managing deadlines
- [x] Managing events

If you Java programmer, you can use it to practice Java too. Here's the `Duke` method:
```
public class Duke {
    public static void main(String[] args) {
        Ui myUi = new Ui();
        myUi.startMessage();
        Storage myStorage = new Storage();
        TaskList myTasks = new TaskList(myStorage.setUp());
        myUi.getUserCommand(myTasks);
    }
}
```

Commit:
Add Gradle support

mqchua committed:
[Level-1]: Done

mqchua committed:
[Level-2]: Done

mqchua committed:
[Level-3]: Done

mqchua committed:
[Level-4]: Done

mqchua committed:
[A-TextUiTesting]: Done

mqchua committed:
[Level-6]: Done

mqchua committed:
Updated test cases

mqchua committed:
Level-7 Branch

mqchua committed:
[Level-8]: Done

mqchua committed:
Merge branch 'add-gradle-support' of https://github.com/nus-cs2103-AY2223S1/ip

mqchua committed:
[A-Gradle]: Done

mqchua committed:
[A-MoreOOP]: Done

mqchua committed:
[Level-8]: Remote branch

mqchua committed:
[A-Packages]: Done

mqchua committed:
[A-JUnit]: Done

mqchua committed:
[A-JavaDoc]: Done

mqchua committed:
[A-CodingStandard]: Done

mqchua committed:
[Level-9]: Done

mqchua committed:
Fix conflicts

mqchua committed:
[Level-8]: Small fix

GenFusion122 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,73 @@
+    void getUserCommand(TaskList myTasks) {
+        Scanner scanner = new Scanner(System.in);
+        while (scanner.hasNextLine()) {
+            String userInput = scanner.nextLine();
```
setting the input to uppercase letters and parsing the uppercased commands like "BYE" could help handle cases where users input commands like "Bye", "bYe", etc..

GenFusion122 commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,73 @@
+                myTasks.addTask(new Deadline(
+                    userInput.substring(9).split("/by")[0], LocalDate.parse(
+                    userInput.split("/by ")[1])));
+                myTasks.saveTasks();
```
does the code handle for events like a user inputting "deadline" only?

GenFusion122 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,120 @@
+    }
+
+    /**
+     * Adds new task to master list
```
Javadocs should contain @params for stuff like myTask

GenFusion122 reviewed (commented):
The code looks great, I've left a few suggestions and some slight nitpicks, but overall great job!

lleyi0606 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,76 @@
+     */
+    public void parser(String line) {
+        String taskType = Character.toString(line.charAt(1));
+        switch (taskType){
```
There should be no indentations for `case` clauses!

lleyi0606 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,120 @@
+
+public class TaskList {
+
+    private final ArrayList<Task> myList;
```
Might be better to name this ArrayList as `tasks` instead because plural to be used on collection of objects

mqchua committed:
Add basic GUI features

mqchua committed:
[Level-10]: Done

mqchua committed:
[Level-10]: Minor fix

mqchua committed:
[A-Assertions]: Done

mqchua committed:
[A-CodeQuality]: Done

mqchua committed:
Add help command in Ui

Gives user more idea on how the product works

Lists the collection of commands that the user will be able to use.
For example, commands such as 'list' that are commonly used are added.

mqchua committed:
Merging PR

mqchua committed:
Merge pull request #2 from mqchua/branch-Merging-PRs

Merging PR

mqchua committed:
Update Readme

mqchua committed:
Update Readme 2

mqchua committed:
Update Readme 3

mqchua committed:
Update readme

mqchua committed:
Add Ui.png
